distribution/docs/src/main/resources/_contents/extending-contents.adoc

```diff
@@ -863,6 +863,132 @@ i.e., without this data, the new system will still have the same configuration a
 Any system related configuration and settings should be handled by a different service that implements the
 `ConfigurationMigratable` interface, not by implementors of this class.
 
+=== Developing CometD Clients for Asynchronous Search and Retrieval
+
+${ddf-branding} uses the http://cometd.org/[CometD] framework to perform asynchronous operations on the catalog.
+This is most apparent in the UI application, which is able to perform multiple queries and display them asynchronously as the results are returned.
+CometD has a http://docs.cometd.org/2/reference/[comprehensive manual] that details how to interact with and best use the framework with a variety of clients (e.g., javascript, java, perl, and python).
+
+The SearchUI code can be used as reference implementation of how a JavaScript client can be created to integrate with the CometD endpoint.
+Examples are included on this page for convenience.
```

Comment:
Is there a guarantee that the examples offered (below) will render on the same page? 
If not, I would consider rewording this to `Examples are included in the following section/this section for convenience.`